dps310/dps310.go

```diff
@@ -0,0 +1,539 @@
+
+	d.c20 = decVal((uint16_t(coeffs[12]) << 8) |
+		uint16_t(coeffs[13]))
+
+	d.c21 = decVal((uint16_t(coeffs[14]) << 8) |
+		uint16_t(coeffs[15]))
+
+	d.c30 = decVal((uint16_t(coeffs[16]) << 8) |
+		uint16_t(coeffs[17]))
+
+	return
+}
+
+type uint12_t uint32
+type uint16_t uint32
+type uint20_t uint32
+type uint24_t uint32
+
+type coefs interface {
+	uint12_t | uint16_t | uint20_t | uint24_t
+}
+
+func decVal[T coefs](v T) float32 {
+	const (
+		max12 = 1<<11 - 1
+		sub12 = 1 << 12
+		max16 = 1<<15 - 1
+		sub16 = 1 << 16
+		max20 = 1<<19 - 1
+		sub20 = 1 << 20
+		max24 = 1<<23 - 1
+		sub24 = 1 << 24
+	)
+
+	var (
+		max, sub int32
+	)
+
+	switch any(v).(type) {
+	case uint12_t:
+		max = max12
+		sub = sub12
+	case uint16_t:
+		max = max16
+		sub = sub16
+	case uint20_t:
+		max = max20
+		sub = sub20
+	case uint24_t:
+		max = max24
+		sub = sub24
+	}
+
+	val := int32(v)
+	if val > max {
+		val -= sub
+	}
+	return float32(val)
+}
+
+// Set the sample rate and oversampling averaging for pressure and temperature
+// rate How many samples per second to take
+// param os How many oversamples to average
+func (d *DPS310) ConfigureValues(rate Rate, os Precision) {
+	d.ConfigurePressure(rate, os)
+	d.ConfigureTemperature(rate, os)
+}
+
+func (d *DPS310) isOkRateOs() bool {
+	return float32(d.pressureRate)*osScalingData[d.pressureOs].measTime+
+		float32(d.tempRate)*osScalingData[d.tempOs].measTime >= 1.0
+}
+
+// Set the sample rate and oversampling averaging for pressure
+// rate How many samples per second to take
+// param os How many oversamples to average
+func (d *DPS310) ConfigurePressure(rate Rate, os Precision) (err error) {
+	// check rate oversamling combination
+	d.pressureRate = rate
+	d.pressureOs = os
+	if !d.isOkRateOs() {
+		return fmt.Errorf("invalid rate oversampling comination")
+	}
+
+	data := (uint8(rate) & PRS_CFG_PM_RATE_MSK)<<4 | 
+	(uint8(os) & PRS_CFG_PM_PRC_MSK)
+	err = d.writeRegister(PRS_CFG, data)
+	if err != nil {
+		return
+	}
+
+	data, err = d.readRegister(CFG_REG)
+	if err != nil {
+		return
+	}
+
+	if os > PRC_8SAMPLES {
+		err = d.writeRegister(CFG_REG, data | CFG_REG_PRS_SHIFT_EN_MSK)
+	} else {
+		err = d.writeRegister(CFG_REG, data & CFG_REG_PRS_SHIFT_DIS_MSK)
+	}
+	if err != nil {
+		return
+	}
+
+	d.pressureScale = osScalingData[os].factor
+
+	return
+}
+
+// Set the sample rate and oversampling averaging for temperature
+// param rate How many samples per second to take
+// param os How many oversamples to average
+func (d *DPS310) ConfigureTemperature(rate Rate, os Precision) (err error) {
+	// find temperatur src
+	var (
+		data uint8
+	)
+	data, err = d.readRegister(COEF_SRCE)
+	data &= COEF_SRCE_TMP_COEF_SR_MSK
+
+	// check rate oversamling combination
+	d.tempRate = rate
+	d.tempOs = os
+	if !d.isOkRateOs() {
+		return fmt.Errorf("invalid rate oversampling comination")
+	}
+
+	// add temp rate and oversampling
+	data += (uint8(rate)&TMP_CFG_TM_PRC_MSK)<<4 | (uint8(os) & TMP_CFG_TM_RATE_MSK)
```

Comment:
```suggestion
	data |= (uint8(rate)&TMP_CFG_TM_PRC_MSK)<<4 | (uint8(os) & TMP_CFG_TM_RATE_MSK)
```
to make the intention more clear here?